src/main/java/restaurant_screens/DeleteRestaurantView.java

```diff
@@ -73,14 +90,15 @@ public DeleteRestaurantView(RestaurantDeleteController restaurantDeleteControlle
     @Override
     public void actionPerformed(ActionEvent e) {
         try {
+            // initialize the presenter, controller and use case interactor
+            RestaurantDeletePresenter presenter = new RestaurantDeleteResponseFormatter();
+            RestaurantDeleteInputBoundary interactor = new deleteRestaurant(restaurantGateway, userGateway, presenter);
+            RestaurantDeleteController restaurantController = new RestaurantDeleteController(interactor);
+            // Check if the person clicked confirm, if so interact
```

Comment:
Looks good, this will be much cleaner than initializing all this in RestaurantView